File: Languages_LinkedList/LinkedList.go
package main

import "fmt"

type node struct {
	value int
	next *node
} 

var list *node
var amount_of_node int

func main() {
	fmt.Print("Enter amount of nodes: ")
	fmt.Scan(&amount_of_node)

	for i := 0; i < amount_of_node; i++ {
		var n = new(node)
		fmt.Print("Node at ", i, ": ")
		fmt.Scan(&n.value)
		n.next = list
		list = n
	}

	ptr := list
	for ptr != nil {
		fmt.Print(" ", ptr.value)
		ptr = ptr.next
	}


}

File: Languages_LinkedList/LinkedList.c
#include <stdio.h>
#include <stdlib.h>

typedef struct node {
    int value;
    struct node *next;
} node;

int main(void) {
    node *list = NULL;
    int amount_of_nodes = 0;

    printf("Enter amount of nodes: ");
    scanf("%i", &amount_of_nodes);

    for (int i = 0; i < amount_of_nodes; i++) {
        node *n = malloc(sizeof(node));

        printf("Enter at node %i: ", i);
        scanf("%i", &n->value);
        n->next = list;
        list = n;
    }

    for (node *ptr = list; ptr != NULL; ptr = ptr->next) {
        printf(" %i", ptr->value);
    }
}
